cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,595 @@
+                unversionedEntry = TableEntry.unversioned(unversionedEntry.getKey().getKey(), segment );
+                tableExtension.put(tableSegment, Collections.singletonList(unversionedEntry), TIMEOUT).join();
+            } else {
+                tableExtension.remove(tableSegment, Collections.singletonList(unversionedEntry.getKey()), TIMEOUT);
+            }
+        }
+    }
+
+    private MetadataStore.SegmentInfo resetStorageSegment(MetadataStore.SegmentInfo segmentInfo) {
+        StreamSegmentInformation segmentInformation = StreamSegmentInformation.builder()
+                .name(segmentInfo.getProperties().getName())
+                .attributes(segmentInfo.getProperties().getAttributes())
+                .build();
+        return MetadataStore.SegmentInfo.builder()
+                .properties(segmentInformation)
+                .segmentId(segmentInfo.getSegmentId())
+                .build();
+    }
+
+    private MetadataStore.SegmentInfo resetSegmentID(MetadataStore.SegmentInfo segmentInfo) {
+        MetadataStore.SegmentInfo resetSegmentInfo = MetadataStore.SegmentInfo.builder()
+                .segmentId(NO_STREAM_SEGMENT_ID)
+                .properties(segmentInfo.getProperties())
+                .build();
+        return resetSegmentInfo;
+    }
+
+    /**
+     * Writes all the parsed TableSegment operations from raw chunks into storage_metadata of the debug
+     * segment container.
+     * @param container Container whose storage_metadata table segment is populated.
+     * @param tableSegment name of container metadata segment.
+     * @param tableSegmentOperations Operations parsed from storage metadata raw chunk file bytes for container metadata.
+     * @throws Exception
+     */
+    private void writeEntriesToStorageMetadata(DebugStreamSegmentContainer container, String tableSegment, List<TableSegmentUtils.TableSegmentOperation> tableSegmentOperations) throws Exception {
+        output("Writing entries to storage metadata");
+        ContainerTableExtension tableExtension = container.getExtension(ContainerTableExtension.class);
+        for (TableSegmentUtils.TableSegmentOperation operation : tableSegmentOperations) {
+            TableSegmentEntry entry = operation.getContents();
+            TableEntry unversionedEntry = TableEntry.unversioned(new ByteBufWrapper(entry.getKey().getKey()), new ByteBufWrapper(entry.getValue()));
+            String segment = new String(unversionedEntry.getKey().getKey().getCopy());
+            if (!allowSegment(segment)) {
+                continue;
+            }
+            if (operation instanceof TableSegmentUtils.PutOperation) {
+                tableExtension.put(tableSegment, Collections.singletonList(unversionedEntry), TIMEOUT).join();
+            } else {
+                tableExtension.remove(tableSegment, Collections.singletonList(unversionedEntry.getKey()), TIMEOUT);
+            }
+        }
+    }
+
+    private boolean allowSegment(String segmentName) {
```

Comment:
I'm not sure I really understood why we exclude EVENT_PROCESSEOR_SEGMENT or scaleGroup.
Please add comment here.

Reply:
added comments.